crates/ruff/tests/direxist_guard.rs

```diff
@@ -0,0 +1,30 @@
+//! Test to verify Ruff's behavior when run from deleted directory.
+//! It has to be isolated in a separate module.
+//! Tests in the same module become flaky under `cargo test`s parallel execution
+//! due to in-test working directory manipulation.
+
+#![cfg(not(target_family = "wasm"))]
```

Comment:
I think the Windows CI failure is real. I think Windows will keep the directory around as long as a process is using it. Maybe we need to exclude both `wasm` and `windows` targets? I always forget the syntax for this, but I think it's something like this:

```suggestion
#![cfg(not(any(target_family = "wasm", target_family = "windows")))]
```

Reply:
I'll leave the Windows CI failures to you, if that's fine... I don't have access to a Windows machine.